src/core_ocean/shared/mpas_ocn_thick_ale.F

```diff
@@ -151,7 +151,8 @@ subroutine ocn_ALE_thickness(meshPool, verticalMeshPool, SSH, ALE_thickness, err
       !
       if (configALEthicknessProportionality==1) then ! restingThickness_times_weights
 
-         !$omp do schedule(runtime)
+         !$omp parallel
+         !$omp do schedule(runtime) private(kMax, thicknessSum, k, SSH_ALE_Thickness)
```

Comment:
This section is producing an error on grizzly using 
```
module load intel/17.0.1 openmpi/1.10.5 netcdf/4.4.1 parallel-netcdf/1.5.0 pio/1.7.2
make ifort CORE=ocean DEBUG=true OPENMP=true GEN_F90=true
```
It looks like iCell is missing in the private list. I added it, and still get this same error:
```
forrtl: severe (174): SIGSEGV, segmentation fault occurred
Image              PC                Routine            Line        Source
...
ocean_model        0000000001B89B2B  ocn_thick_ale_mp_         172  mpas_ocn_thick_ale.f90
```
which is this line, just below:
```
160                SSH_ALE_Thickness(k) = SSH(iCell) * vertCoordMovementWeights(k) * restingThickness(k, iCell)
```
I have not found the root problem yet. @mattdturner if your tests on cori don't reproduce it, it might be an intel version thing. Could you double check that the allocatable array `SSH_ALE_Thickness(k)` is declared correctly? It does not need to be module level with these changes?

I can confirm that this error does not occur with optimization on, or with `OPENMP=false`.

Reply:
As the code currently stands, I should be seeing the error that you are seeing.  I'm not sure why it didn't show up in my test on Cori, but the `SSH_ALE_Thickness` array is uninitialized within the OpenMP region.  I'll look into this now.

Reply:
I just pushed a commit that changes `SSH_ALE_Thickness` to `firstprivate`, and I was able to successfully run a test on Grizzly.